### ISSUE TITLE
custom-error-definition doesn't report error if the Error class is exported with `module.exports`

### ISSUE THREAD

**Issue by pvdlg** [sindresorhus/eslint-plugin-unicorn#127]
This code report the error ``The `name` property should be set `FooError`.`` as expected:
```js
class FooError extends Error {
  constructor() {
    super('My awesome Foo Error');
    this.name = this.constructor.name;
  }
};
```

This code doesn't report any error:
```js
module.exports = class FooError extends Error {
  constructor() {
    super('My awesome Foo Error');
    this.name = this.constructor.name;
  }
};
```

With `export` it works as expected and the error is reported:
```js
export class FooError extends Error {
  constructor() {
    super('My awesome Foo Error');
    this.name = this.constructor.name;
  }
}
```
